### PR TITLE
Feature/reset gps ref service

### PR DESCRIPTION
**Pull request by raghavkhanna**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,47 +1,20 @@
...
+find_package(catkin_simple REQUIRED)
+catkin_simple(ALL_DEPS_REQUIRED)
 
+find_package(Eigen REQUIRED)
+include_directories(${Eigen_INCLUDE_DIRS})
 
-add_executable(set_gps_reference_node src/set_gps_reference_node.cpp)
-add_dependencies(set_gps_reference_node ${catkin_EXPORTED_TARGETS})
+cs_add_executable(set_gps_reference_node src/set_gps_reference_node.cpp)
 target_link_libraries(set_gps_reference_node  ${catkin_LIBRARIES})
```
Comment (enricgalceran) [inline CMakeLists.txt:13]: I'm not clear on whether target_link_libraries is necessary for just a binary, with no libraries. Please compare with other nodes (though if it compiles cleanly it should be OK).

File: src/set_gps_reference_node.cpp [hunk 2]
```diff
@@ -2,10 +2,13 @@
 #include <sensor_msgs/NavSatFix.h>
 #include <string>
 #include <vector>
+#include <std_srvs/Empty.h>
 
 double g_lat_ref;
 double g_lon_ref;
 double g_alt_ref;
+int count = 1;
```
Comment (enricgalceran) [inline src/set_gps_reference_node.cpp]: Rename to g_count.

File: src/set_gps_reference_node.cpp [hunk 3]
```diff
@@ -2,10 +2,13 @@
...
+bool gps_ref_is_init;
 
 std::vector<double> g_pressure_heights;
```
Comment (enricgalceran) [inline src/set_gps_reference_node.cpp]: Not needed anymore. Please remove.

File: src/set_gps_reference_node.cpp [hunk 4]
```diff
@@ -18,49 +21,63 @@ enum EMode
...
+      nh.setParam("/gps_ref_is_init", true);
 
-    ros::shutdown();
-    return;
-  } else {
-    ROS_INFO("    Still waiting for %d measurements", g_its - count);
-  }
+      ROS_INFO("Final reference position: %f, %f, %f", g_lat_ref, g_lon_ref, g_alt_ref);
+
+      //ros::shutdown();
```
Comment (enricgalceran) [inline src/set_gps_reference_node.cpp]: Remove.

**Comment by enricgalceran** [comment]
See minor changes. Otherwise LGTM.

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,47 +1,20 @@
...
 
-add_executable(gps_to_pose_conversion_node src/gps_to_pose_conversion_node.cpp)
-add_dependencies(gps_to_pose_conversion_node ${catkin_EXPORTED_TARGETS})
+cs_add_executable(gps_to_pose_conversion_node src/gps_to_pose_conversion_node.cpp)
 target_link_libraries(gps_to_pose_conversion_node  ${catkin_LIBRARIES})
 
-
+cs_install()
+cs_install_scripts(src/gps_spoofer.py src/euler_radians_to_degrees.py src/odometry_height_adjuster.py)
```
Comment (marija-p) [inline CMakeLists.txt]: Can you please remove "odometry_height_adjuster.py" from the scripts to install? We are removing this node as it is unused.

Comment (marija-p) [inline CMakeLists.txt]: Actually, I'll take care of this in my PR. Nevermind. ;)

Comment (marija-p) [inline CMakeLists.txt]: ... My PR is actually using a version of geodetic_utils without catkin_simple, so I cannot remove this. Can you take care of it @raghavkhanna ?

File: CMakeLists.txt (hunk 1 above)
Comment (raghavkhanna) [inline CMakeLists.txt:13]: removed target_link_libraries, not neccessary

**Pull request merged**